pkg/restarter/restarter_test.go

```diff
@@ -0,0 +1,303 @@
+	}
+	deps.Namespace = metav1.NamespaceDefault
+	stopCh := make(chan struct{})
+
+	e := newEndpoint(deps.Service, deps.Namespace, deps.Labels)
+	pC := newPodInCrashloop("pod-0", map[string]string{
+		"garden.sapcloud.io/role": "controlplane",
+		"role":                    "NotEtcd",
+	})
+	pH := newPodHealthy("pod-1", map[string]string{
+		"garden.sapcloud.io/role": "controlplane",
+		"role":                    "NotEtcd",
+	})
+
+	f.eLister = append(f.eLister, e)
+	//f.pLister = append(f.pLister, p)
+	f.objects = append(f.objects, e, pC, pH)
+	f.client = fake.NewSimpleClientset(f.objects...)
+	watcher := watch.NewFake()
+	f.client.PrependWatchReactor("pods", test.DefaultWatchReactor(watcher, nil))
+	// simulate add/update/delete watch events
+
+	c, _, err := f.newController(deps, stopCh)
+	if err != nil {
+		t.Fatalf("error creating Deployment controller: %v", err)
+	}
+	go func() {
+		pl, err := f.client.CoreV1().Pods(metav1.NamespaceDefault).List(metav1.ListOptions{})
+		if err != nil {
+			t.Fatalf("error fetching pods: %v", err)
+		}
+		before := len(pl.Items)
+		t.Logf("Number of pods before the watchdog started: %v.", before)
+		watcher.Add(pC)
+		watcher.Add(pH)
+		pl, err = f.client.CoreV1().Pods(metav1.NamespaceDefault).List(metav1.ListOptions{})
+		if err != nil {
+			t.Fatalf("error fetching pods: %v", err)
+		}
+		after := len(pl.Items)
+		t.Logf("Number of pods after the watchdog started: %v.", after)
+		close(stopCh)
+		if before == after {
+			t.Error("Pod in CrashloopBackoff not deleted by the dependency-watchdog.")
+		}
+	}()
+
+	t.Logf("Starting dep watchdog.\n")
+	c.Run(1)
+
+}
+
+func TestDeletePodTransitioningToCrashloopBackoff(t *testing.T) {
+	f := newFixture(t)
+	deps, err := DecodeConfigFile([]byte(dep))
+	if err != nil {
+		t.Fatalf("error decoding file: %v", err)
+	}
+	deps.Namespace = metav1.NamespaceDefault
+	stopCh := make(chan struct{})
+
+	e := newEndpoint(deps.Service, deps.Namespace, deps.Labels)
+	// pC := newPodInCrashloop("pod-0", map[string]string{
+	// 	"garden.sapcloud.io/role": "controlplane",
+	// 	"role":                    "NotEtcd",
+	// })
+	pH := newPodHealthy("pod-1", map[string]string{
+		"garden.sapcloud.io/role": "controlplane",
+		"role":                    "NotEtcd",
+	})
+
+	f.eLister = append(f.eLister, e)
+	//f.pLister = append(f.pLister, p)
+	f.objects = append(f.objects, e, pH)
+	f.client = fake.NewSimpleClientset(f.objects...)
+	watcher := watch.NewFake()
+	f.client.PrependWatchReactor("pods", test.DefaultWatchReactor(watcher, nil))
+	// simulate add/update/delete watch events
+
+	c, _, err := f.newController(deps, stopCh)
+	if err != nil {
+		t.Fatalf("error creating Deployment controller: %v", err)
+	}
+	go func() {
+		pl, err := f.client.CoreV1().Pods(metav1.NamespaceDefault).List(metav1.ListOptions{})
+		if err != nil {
+			t.Fatalf("error fetching pods: %v", err)
+		}
+		before := len(pl.Items)
+		t.Logf("Number of pods before the watchdog started: %v.", before)
+		watcher.Add(pH)
```

Comment:
Why not use the already existing mechanism in MCM tests? Even dispatching initial objects has been [implemented](https://github.com/gardener/machine-controller-manager/blob/master/pkg/fakeclient/client.go#L297:19).